frame/support/src/traits/tokens/nonfungibles_v2.rs

```diff
@@ -330,12 +330,26 @@ pub trait Mutate<AccountId, ItemConfig>: Inspect<AccountId> {
 	}
 }
 
-/// Trait for transferring non-fungible sets of items.
+/// Trait for transferring and locking/unlocking non-fungible sets of items.
 pub trait Transfer<AccountId>: Inspect<AccountId> {
 	/// Transfer `item` of `collection` into `destination` account.
 	fn transfer(
 		collection: &Self::CollectionId,
 		item: &Self::ItemId,
 		destination: &AccountId,
 	) -> DispatchResult;
+
+	/// Disable the `item` of `collection` transfer.
+	///
+	/// By default, this is not a supported operation.
+	fn lock(_collection: &Self::CollectionId, _item: &Self::ItemId) -> DispatchResult {
+		Err(TokenError::Unsupported.into())
```

Comment:
q: should this be introduced to nonfungibles v1?

Reply:
I don't think it should be on this trait. I would have nft lock/unlock in a separate trait.

Reply:
If we must have a single trait then can we have a `LockableNonfungible` trait, leave `Transfer` unchanged and have a `TransferLockable` trait that inherits from both?

Reply:
> I don't think it should be on this trait. I would have nft lock/unlock in a separate trait.

I like the idea of having `Transfer` and `LockableNonfungible ` separate. Especially if `lock` is disabled by default and not every NFT is lockable by design

Reply:
Well, similar functionality for Fungibles lives in ‘Transfer’ trait, this seems consistent to me as opposed to making them diverge further. That way somebody investigating fungibles code could have an easier time finding their way around non-fungibles and vice versa.
Not to mention the fact that lock/unlock directly affects the transferability.